### PR TITLE
FinEID v4 support

### PR DESCRIPTION
**Pull request by metsma**
WE2-751

Signed-off-by: Raul Metsma <raul@metsma.ee>

### CODE REVIEW

File: src/electronic-ids/pcsc/FinEID.cpp [hunk 1]
```diff
@@ -99,9 +98,9 @@ ElectronicID::PinRetriesRemainingAndMax FinEIDv3::signingPinRetriesLeftImpl() co
 }
 
 byte_vector FinEIDv3::sign(const HashAlgorithm hashAlgo, const byte_vector& hash,
-                           const byte_vector& pin, byte_vector::value_type pinReference,
-                           PinMinMaxLength pinMinMaxLength, byte_vector::value_type keyReference,
-                           byte_vector::value_type signatureAlgo, byte_vector::value_type LE) const
+                           const byte_vector& pin, value_type pinReference,
```
Comment (mrts) [inline src/electronic-ids/pcsc/FinEID.cpp:101]: To my eyes `value_type` is ambiguous - when I look at this code, it makes me ask "what value_type?". Let's use `byte_type`?

File: src/electronic-ids/pcsc/FinEID.hpp [hunk 2]
```diff
@@ -32,8 +32,11 @@ class FinEIDv3 : public PcscElectronicID
 public:
     FinEIDv3(pcsc_cpp::SmartCard::ptr _card) : PcscElectronicID(std::move(_card)) {}
 
-private:
-    pcsc_cpp::byte_vector getCertificateImpl(const CertificateType type) const override;
+protected:
+    using byte_vector = pcsc_cpp::byte_vector;
+    using value_type = byte_vector::value_type;
```
Comment (mrts) [inline src/electronic-ids/pcsc/FinEID.hpp:37]: `FinEIDv3::value_type` reads "the value type of FinEIDv3". In my opinion "the byte type of FinEIDv3" is more correct:
```suggestion
    using byte_type = byte_vector::value_type;
```

File: src/electronic-ids/pcsc/pcsc-common.hpp [hunk 3]
```diff
@@ -177,4 +177,21 @@ inline pcsc_cpp::byte_vector computeSignature(pcsc_cpp::SmartCard& card,
     return response.data;
 }
 
+inline void selectComputeSignatureEnv(pcsc_cpp::SmartCard& card,
+                                      pcsc_cpp::byte_vector::value_type signatureAlgo,
+                                      pcsc_cpp::byte_vector::value_type keyReference,
+                                      const std::string& cardType)
+{
+    static const pcsc_cpp::CommandApdu COMPUTE_SIGNATURE {0x00, 0x22, 0x41, 0xB6};
```
Comment (mrts) [inline src/electronic-ids/pcsc/pcsc-common.hpp:185]:
```suggestion
    static const pcsc_cpp::CommandApdu SET_COMPUTE_SIGNATURE_ENV {0x00, 0x22, 0x41, 0xB6};
```

**Review by mrts — CHANGES_REQUESTED** [review]

File: src/electronic-ids/pcsc/FinEID.cpp (hunk 1 above)
Comment (metsma) [reply src/electronic-ids/pcsc/FinEID.cpp:101]: I just created alias to shorten code. byte_type should then be declared in libpcsc-cpp

File: src/electronic-ids/pcsc/FinEID.hpp (hunk 2 above)
Comment (metsma) [reply src/electronic-ids/pcsc/FinEID.hpp:37]: It is just alias tho shorten. Please define byte type in pcsc-cpp

Comment (mrts) [reply src/electronic-ids/pcsc/FinEID.hpp:37]: Will do, thanks for the suggestion!

**Review by mrts — APPROVED** [review]

**Pull request merged**